### ISSUE TITLE
object-shorthand rule fails with quoted properties

### ISSUE THREAD

**Issue by kentor** [eslint/eslint#2374]
.eslintrc:
```yaml
---
rules:
  object-shorthand: [2, always]
  no-unused-vars: 0
```

index.js:
```js
var o = { "a": "a" };
```

```bash
index.js
  1:10  error  Expected property shorthand  object-shorthand

✖ 1 problem (1 error, 0 warnings)
```

Shouldn't quoted properties only concern the `quote-props` rule?

**Comment by nzakas** [comment]
Quoted properties should still be checked, because this is redundant:

```js
var o = { "a": a };
```

But literal values definitely should be excluded. @xjamundx do you want to take a look at this?

**Comment by xjamundx** [comment]
Yup will try to fix tonight, otherwise early next week!
